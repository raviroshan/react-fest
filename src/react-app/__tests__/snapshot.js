import React from 'react';
import ReactDOM from 'react-dom';
import { getProductsBycategory } from '../../../test/snapshot-mock';

const renderer = require('react-test-renderer');

test('manual "snapshot"', () => {
  const products = getProductsBycategory();
  expect(products).toEqual([
    { name: 'product1', Price: 100, category: ['food'] },
    { name: 'product7', Price: 700, category: ['food'] },
  ]);
});

test('automatic snapshot', () => {
  const products = getProductsBycategory();
  expect(products).toMatchSnapshot();

  // DOM NODES!!!
  const div = document.createElement('div');
  const title = '<h2 class="title">Product List</h2>';
  expect(div).toMatchSnapshot('title of a snapshot!');


  // And react elements!
  const onClick = () => {};
  const element = React.createElement('div', { onClick }, 'Hello World');
  expect(element).toMatchSnapshot('react element');


  // and rendered elements
  const rendered = renderer.create(element);
  expect(rendered).toMatchSnapshot('rendered');

  // and DOM nodes rendered via react
  const app = document.createElement('div');
  ReactDOM.render(element, app);
  expect(app).toMatchSnapshot('react-dom');
});
