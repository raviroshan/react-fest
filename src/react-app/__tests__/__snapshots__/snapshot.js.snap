// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automatic snapshot 1`] = `
Array [
  Object {
    "Price": 100,
    "category": Array [
      "food",
    ],
    "name": "product1",
  },
  Object {
    "Price": 700,
    "category": Array [
      "food",
    ],
    "name": "product7",
  },
]
`;

exports[`automatic snapshot: react element 1`] = `
<div
  onClick={[Function]}
>
  Hello World
</div>
`;

exports[`automatic snapshot: react-dom 1`] = `
<div>
  <div>
    Hello World
  </div>
</div>
`;

exports[`automatic snapshot: rendered 1`] = `
<div
  onClick={[Function]}
>
  Hello World
</div>
`;

exports[`automatic snapshot: title of a snapshot! 1`] = `<div />`;
